hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -107,7 +108,11 @@ public HoodieFileGroupReader(HoodieReaderContext<T> readerContext,
                                HoodieTableConfig tableConfig,
                                long start,
                                long length,
-                               boolean shouldUseRecordPosition) {
+                               boolean shouldUseRecordPosition,
+                               long maxMemorySizeInBytes,
+                               String spillableMapBasePath,
```

Comment:
a bit ugly though. Can we wrap these parameters into a class first?

Reply:
yeah, there are two many parameters, we should add a builder for it just like what we do to `AbstractHoodieLogRecordReader`.